src/ProjectSelectionManager.cpp

```diff
@@ -47,88 +47,51 @@ const ProjectSelectionManager &ProjectSelectionManager::Get(
    return Get( const_cast< AudacityProject & >( project ) );
 }
 
-ProjectSelectionManager::ProjectSelectionManager( AudacityProject &project )
-   : mProject{ project }
-{
-}
-
-ProjectSelectionManager::~ProjectSelectionManager() = default;
-
-bool ProjectSelectionManager::SnapSelection()
-{
-   auto &project = mProject;
-   auto &formats = ProjectNumericFormats::Get( project );
-   auto &settings = ProjectSettings::Get( project );
-   auto &window = ProjectWindow::Get( project );
-   auto snapTo = settings.GetSnapTo();
-   if (snapTo != SNAP_OFF) {
-      auto &viewInfo = ViewInfo::Get( project );
-      auto &selectedRegion = viewInfo.selectedRegion;
-      NumericConverter nc(NumericConverter::TIME,
-         formats.GetSelectionFormat(), 0, ProjectRate::Get(project).GetRate());
-      const bool nearest = (snapTo == SNAP_NEAREST);
-
-      const double oldt0 = selectedRegion.t0();
-      const double oldt1 = selectedRegion.t1();
-
-      nc.ValueToControls(oldt0, nearest);
-      nc.ControlsToValue();
-      const double t0 = nc.GetValue();
-
-      nc.ValueToControls(oldt1, nearest);
-      nc.ControlsToValue();
-      const double t1 = nc.GetValue();
-
-      if (t0 != oldt0 || t1 != oldt1) {
-         selectedRegion.setTimes(t0, t1);
-         return true;
-      }
-   }
+ProjectSelectionManager::ProjectSelectionManager(AudacityProject& project)
+    : mProject { project }
+    , mSnapModeChangedSubscription { ProjectSnap::Get(project).Subscribe(
+         [this](auto&)
+         {
+            SnapSelection();
+         })}
+    , mTimeSignatureChangedSubscription {
+       ProjectTimeSignature::Get(project).Subscribe([this](auto&)
+                                                    { SnapSelection(); })
+    }
 
-   return false;
-}
-
-double ProjectSelectionManager::AS_GetRate()
 {
-   return ProjectRate::Get(mProject).GetRate();
 }
 
-void ProjectSelectionManager::AS_SetRate(double rate)
-{
-   auto &project = mProject;
-   ProjectRate::Get( project ).SetRate( rate );
-}
+ProjectSelectionManager::~ProjectSelectionManager() = default;
 
-int ProjectSelectionManager::AS_GetSnapTo()
-{
-   auto &project = mProject;
-   auto &settings = ProjectSettings::Get( project );
-   return settings.GetSnapTo();
-}
+void ProjectSelectionManager::SnapSelection()
+{   
+   auto& project = mProject;
+   auto& projectSnap = ProjectSnap::Get(mProject);
 
-void ProjectSelectionManager::AS_SetSnapTo(int snap)
```

Comment:
This function is removed, but there is still one comment mentioning it.  Remove that comment.